docs: provision a static local volume (#4163)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.1/storage/static-local-volume/index.md

---
layout: layout.pug
navigationTitle: Provision a static local volume
title: Provision a static local volume
menuWeight: 15
excerpt: Learn how to provision a static local volume for a Konvoy cluster
beta: false
enterprise: false
---

The `localvolumeprovisioner` component uses the [local volume static provisioner][localstorage] to manage persistent volumes for pre-allocated disks.
It does this by watching the `/mnt/disks` folder on each host and creating persistent volumes in the `localvolumeprovisioner` storage class for each disk that it discovers in this folder.

> -   Persistent volumes with a 'Filesystem' volume-mode are discovered if they are mounted under `/mnt/disks`.

Reviewer: ```suggestion
-   Persistent volumes with a 'Filesystem' volume-mode are discovered if you mount them under `/mnt/disks`.
```

> -   Persistent volumes with a 'Block' volume-mode are discovered if a symbolic link to the block device is created in `/mnt/disks`.

Reviewer: ```suggestion
-   Persistent volumes with a 'Block' volume-mode are discovered if you create a symbolic link to the block device in `/mnt/disks`.
```

## Before you begin

Before starting this tutorial, verify the following:

-   You have access to a Linux, macOS, or Windows computer with a supported operating system version.

> -   You have a provisioned `konvoy` cluster that uses the `localvolumeprovisioner` platform application, but have not added any other Kommander applications to the cluster yet.

Author: 🤔 Can anyone confirm if this should change yet? I have been recently updating Kommander to dkp in Kommander UI, however, I was told that the docs will still reference Kommander until those changes are fully sorted out. There has been a lot of confusion around this lately.

> -   You have a provisioned `konvoy` cluster that uses the `localvolumeprovisioner` platform application, but have not added any other Kommander applications to the cluster yet.

Reviewer 2: IMO, just leave it for now and we can change it once we get an answer from Deepak.

> -   You have a provisioned `konvoy` cluster that uses the `localvolumeprovisioner` platform application, but have not added any other Kommander applications to the cluster yet.

Author: https://mesosphere.slack.com/archives/C052HL47U/p1647562375547639?thread_ts=1647015731.820569&cid=C052HL47U

Deepak replied with - "There is no more konvoy or kommander cluster. They all are DKP clusters."

For this tutorial, you **do not deploy** using all the default settings as described in the [Quick start guide][quickstart].

This distinction between provisioning and deployment is important because some applications depend on the storage class provided by the `localvolumeprovisioner` component and can fail to start if it is not configured yet.

### Provision the cluster and a volume

1.  Create a pre-provisioned cluster by following the steps outlined [in the choose pre-provisioned infrastructure topic][preprovision].

>     Whenever volumes are create on the node machines, the local volume provisioned detects the volume in `/mnt/disks` and adds it as a persistent volume with the `localvolumeprovisioner` storage class.

Reviewer 3: ```suggestion
    Whenever volumes are created on the machines, the `localvolumeprovisioner` detects the volume in `/mnt/disks` and adds it as a persistent volume with the `localvolumeprovisioner` storage class.
```

> 1.  Create some volumes in `/mnt/disks` on each host.

Reviewer: Is more than one required? If not, we may want to write, "Create at least one volume in...."

If more than one **is** required, how many do I have to have?

Author: @dkoshkin / @faiq  can you confirm the requirements with this?

Reviewer 3: It is required if the user wants that node to run workloads with a volume. 

So, I think the "Create at least one filesystem on each host"

    For example, mount a `tmpfs` volume:

    ```bash
    mkdir -p /mnt/disks/example-volume && mount -t tmpfs example-volume /mnt/disks/example-volume
    ```

1.  Verify the persistent volume by running the following command:

    ```bash
    kubectl get pv
    ```

    The command displays output similar to the following:

    ```sh
    NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS             REASON   AGE
    local-pv-4c7fc8ba   3986Mi     RWO            Delete           Available           localvolumeprovisioner            2s
    ```

> 1.  Claim the persistent volume using PersistentVolumeClaim settings by running the following command:

Reviewer: ```suggestion
1.  Claim the persistent volume using a PersistentVolumeClaim, by running the following command:
```

    ```yaml
    cat <<EOF | kubectl create -f -
    kind: PersistentVolumeClaim
    apiVersion: v1
    metadata:
      name: example-claim
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 100Mi
      storageClassName: localvolumeprovisioner
    EOF
    ```

> 1.  Reference the persistent volume claim in pods by running the following command:

Reviewer: ```suggestion
1.  Reference the persistent volume claim in a pod by running the following command:
```

    ```yaml
    cat <<EOF | kubectl create -f -
    apiVersion: v1
    kind: Pod
    metadata:
      name: pod-with-persistent-volume
    spec:
      containers:
        - name: frontend
          image: nginx
          volumeMounts:
            - name: data
              mountPath: "/var/www/html"
      volumes:
        - name: data
          persistentVolumeClaim:
            claimName: example-claim
    EOF
    ```

1.  Verify the persistent volume claim by running the following command:

    ```bash
    kubectl get pvc
    ```

    The command displays output similar to the following:

    ```sh
    NAME            STATUS   VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS             AGE
    example-claim   Bound    local-pv-4c7fc8ba   3986Mi     RWO            localvolumeprovisioner   78s
    $ kubectl get pv
    NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                   STORAGECLASS             REASON   AGE
    local-pv-4c7fc8ba   3986Mi     RWO            Delete           Bound       default/example-claim   localvolumeprovisioner            15m
    ```

> Upon deletion of the persistent volume claim, the corresponding persistent volume resource deletes (mandated by the "Delete" reclaim policy), which removes all data on the volume.

Reviewer: ```suggestion
Upon deletion of the persistent volume claim, the corresponding persistent volume resource uses the "Delete" reclaim policy, which removes all data on the volume.
```

---
layout: layout.pug
navigationTitle: Provision a static local volume
title: Provision a static local volume
menuWeight: 15
excerpt: Learn how to provision a static local volume for a Konvoy cluster
beta: false
enterprise: false
---

<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->

The `localvolumeprovisioner` addon uses the [local volume static provisioner][localstorage] to manage persistent volumes for pre-allocated disks.
It does this by watching the `/mnt/disks` folder on each host and creating persistent volumes in the `localvolumeprovisioner` storage class for each disk that is discovered in this folder.

- Persistent volumes with a 'Filesystem' volume-mode are discovered if they are mounted under `/mnt/disks`.

- Persistent volumes with a 'Block' volume-mode are discovered if a symbolic link to the block device is created in `/mnt/disks`.

# Before you begin

> Before starting this tutorial, you should verify the following:

Reviewer 4: ```suggestion
Before starting this tutorial, verify the following:
```

- You have access to a Linux, macOS, or Windows computer with a supported operating system version.

> - You have a provisioned `konvoy` cluster that uses the `localvolumeprovisioner` addon, but have not added any other addons to the cluster yet.

Reviewer 3: Is the term addon something we are still using?

Author: @cmastr / @gvasquezvargas do you know about this one?

Reviewer: I think we use the term "platform application" now.

Reviewer 2: yeah don't use addon.

>   For this tutorial, you **should not deploy** using all the default settings as described in the [Quick start][quickstart].

Reviewer 4: ```suggestion
  For this tutorial, **do not deploy** using all the default settings as described in the [Quick start][quickstart].
```

>   This distinction between provisioning and deployment is important because some addons might depend on the storage class provided by the `localvolumeprovisioner` addon and might fail to start if it isn't configured yet.

Reviewer 4: ```suggestion
  This distinction between provisioning and deployment is important because some addons depend on the storage class provided by the `localvolumeprovisioner` addon and can fail to start if it is not configured yet.
```

## Provision the cluster and a volume

> 1. Provision the Kubernetes cluster without deploying addons by running the following command:
> 
>     ```bash
>     konvoy provision
>     ```
> 
> 1. Provision the local volume provisioner to watch for mounts in `/mnt/disks` on each host.
> 
>     For example, mount a `tmpfs` volume with `ansible` using the `inventory.yaml` provided by `konvoy` by running the following command:
> 
>     ```bash
>     ansible -i inventory.yaml node -m shell -a "mkdir -p /mnt/disks/example-volume && mount -t tmpfs example-volume /mnt/disks/example-volume"
>     ```
> 
> 1. Deploy the `konvoy` cluster with addons by running the following command:
> 
>     ```bash
>     konvoy up
>     ```
> 
>     When you run this command, the local volume provisioned detects the `example-volume` volume and adds it as a persistent volume to the `localvolumeprovisioner` storage class.
> 

Reviewer 5: (Please check the formatting still looks good)
```suggestion

1. Create a pre-provisioned cluster by following the steps outlined [here](https://docs.d2iq.com/dkp/konvoy/2.1/choose-infrastructure/pre-provisioned/).

    Whenever volumes are create on the node machines, the local volume provisioned detects the volume in `/mnt/disks` and adds it as a persistent volume with the `localvolumeprovisioner` storage class.

1. Create some volumes in `/mnt/disks` on each host.

    For example, mount a `tmpfs` volume:

    ```bash
    mkdir -p /mnt/disks/example-volume && mount -t tmpfs example-volume /mnt/disks/example-volume
    ```
```

Author: Sorry not sure what happened there. Updating ✨

1. Verify the persistent volume by running the following command:

    ```bash
    kubectl get pv
    ```

    The command displays output similar to the following:

>     ```bash

Reviewer 4: I think we are using bash for commands and sh for output (so the little copy button does not appear next to the code and customers don't try to run output mistakenly) 
```suggestion
    ```sh
```
